connector/rest/src/main/java/it/pagopa/selfcare/dashboard/connector/rest/UserConnectorImpl.java

```diff
@@ -179,6 +179,20 @@ public Collection<UserInfo> getUsers(String institutionId, UserInfo.UserInfoFilt
                 .orElse(Collections.emptyList());
     }
 
+    @Override
+    public List<String> retrieveFilteredUserInstitution(String institutionId, UserInfo.UserInfoFilter userInfoFilter, String loggedUserId) {
+
+        return Optional.ofNullable(userInstitutionApiRestClient._institutionsInstitutionIdUserInstitutionsGet(institutionId,
+                                null,
+                                List.of(userInfoFilter.getProductId()),
+                                null,
+                                userInfoFilter.getAllowedStates().stream().map(Enum::name).toList(),
```

Comment:
you could check if getAllowedStates is not null before call stream to avoid nullpointer

Reply:
AllowedStates is always not null, but I add this check :)